### PR TITLE
CI/CD: Server and subgraph rinkeby on master commits, mainnet on v* tag

### LINKED ISSUE #174: Server: continuously deploy rinkeby instance
We currently have `govern.backend.aragon.org` that is updated on every version tag.
I will add a rinkeby instance `govern-rinkeby.backend.aragon.org` which will deploy continuously on every commit.

### PR DESCRIPTION
**Pull request by promaty**
With these manifests rinkeby subgraph/server will deploy on every master commit and mainnet on v* tag.
Also I moved server `Dockerfile` to the top so we install it using `yarn.lock` file.

closes #174

### CODE REVIEW

File: .github/workflows/server-ci-cd.yml [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
     - v*
 env:
-  # This is a base repository and we use ${GITHUB_REF##*/} to set the version of the container
+  # This is a base repository and we use git refs to set the version of the container
+  # ${GITHUB_REF##*/} will be either a branch name or a tag depending on the event
   REPO: aragon/govern-server
 
 jobs:
   build:
+    if: github.ref == 'refs/heads/master'
```
Comment (kanaagawa) [inline .github/workflows/server-ci-cd.yml]: I wonder if this condition is meet even on a push tag.

File: .github/workflows/subgraph-ci-cd.yml [hunk 2]
```diff
@@ -11,7 +13,28 @@ jobs:
...
+    - name: Install node
+      uses: actions/setup-node@v1
+      with:
+        node-version: 12
+
+    - name: Cache yarn modules
+      uses: actions/cache@v2
+      with:
+        path: '**/node_modules'
+        key: yarn-${{ hashFiles('yarn.lock') }}
```
Comment (kanaagawa) [inline .github/workflows/subgraph-ci-cd.yml:26]: Nice!

**Review by 0xGabi — APPROVED** [review]

File: .github/workflows/server-ci-cd.yml (hunk 1 above)
Comment (promaty) [reply .github/workflows/server-ci-cd.yml:15]: Yes it is. The idea is to deploy to rinkeby on every master commit and to master when you add `v*` tag I think this makes sense?

Comment (promaty) [reply .github/workflows/server-ci-cd.yml]: good point, removed this part to build on tag as well

**Pull request merged**